.gitlab-ci.yml

```diff
@@ -0,0 +1,68 @@
+stages:          # List of stages for jobs, and their order of execution
+  - build-docker
+  - test
+
+workflow:
+  rules:
+    # Only run the pipeline if its for a pull request or the default branch
+    - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
+    - if: '$CI_EXTERNAL_PULL_REQUEST_IID != null || $CI_EXTERNAL_PULL_REQUEST_IID != ""'
+
+docker_build:
+  stage: build-docker
+  image: docker:20.10
+  tags:
+    - gpu
+    - docker
+  variables:
+    # For this to work, $CI_COMMIT_REF_SLUG == $CI_DEFAULT_BRANCH
```

Comment:
I'm not sure I understand this comment. Is this saying we assume this equality, and something will fail if it's not satisfied?

Reply:
Will clarify the comment, the idea is that the commit slug needs to be equivalent to the default branch. IE slugifying the default branch, in our case `master` must result in `master`. This wouldn't be true if we had something like `this/is-silly` would become `thisissilly`.